examples/bigbench_hard/bigbench_hard.py

```diff
@@ -0,0 +1,361 @@
+    logging.info("return_most_frequent_answer: scores: {}".format(scores))
+    frequent_answer = max(scores, key=scores.get)
+    for thought in thoughts:
+        thought.score = scores[thought.state["current"]]
+        if ignore_empty_answers:
+            if thought.state is None:
+                thought.score = 0
+    return thoughts
+
+
+def test_answer(state: Dict) -> bool:
+    logging.debug(f"\nground truth: {state['ground_truth']}\n current_answer: {state['current']}")
+    ground_truth = state["ground_truth"]
+    current_answer = state["current"]
+    return ground_truth == current_answer
+
+
+class BigBenchHardPrompter(prompter.Prompter):
+    """
+    Generate Prompts for the BigBench-Hard Dataset.
+    """
+
+    sys_prompt = """Provide the answer in the exact format as given"""
+    io_prompt = """<Instruction> {instruction} </Instruction>
+    
+<Examples>
+{examples}
+</Examples>
+    
+<Input> {input} </Input>"""
+
+    answer_prompt = """<Answer>{answer}</Answer>"""
+
+    cot_zeroshot_prompt = f"""<Instruction> {{instruction}} </Instruction>
+    
+    <Input> {{input}} </Input>
+    Let us think Step by Step, then provide the answer in this format: {answer_prompt}"""
+
+    def __init__(self, task: str):
+        """
+        @param task: The bigbench task to create prompts for.
+        """
+        self.task = task
+
+    def generate_prompt(self, num_branches: int, original: str, current: str, method: str, **kwargs) -> str:
+        """
+        Generate a generate prompt for the language model.
+
+        :param num_branches: The number of responses the prompt should ask the LM to generate.
+        :type num_branches: int
+        :param original: Input text.
+        :type original: str
+        :param current: Intermediate solution.
+        :type current: str
+        :param method: Method for which the generate prompt is generated.
+        :type method: str
+        :param kwargs: Additional keyword arguments.
+        :return: The generate prompt.
+        :rtype: str
+        :raise AssertionError: If the requested number of branches is not one.
+        """
+
+        if current is None or current == "":
+            input_str = original
+        else:
+            input_str = current
+
+        prompt_path = project.datasets_dir() / "BIG-Bench-Hard" / "cot-prompts" / f"{self.task}.txt"
+        full_prompt: str = ""
+        with open(prompt_path, "r") as f:
+            for _ in range(2):  # to skip canary warning
+                f.readline()
+            prompt: str = f.readline().rstrip("\n")  # prompt is always on the 3rd line for each task
+            examples: [str] = f.read().lstrip("\n").split("\n\n")  # the rest of the file constitutes the example
+            answers: [str] = []
+            for example in examples:
+                answers.append(example.split("So the answer is ")[-1].removesuffix("."))  # extract answer from example
+            full_examples: [str] = []
```

Comment:
please change them from str to List[str].
```suggestion
            examples: List[str] = f.read().lstrip("\n").split("\n\n")  # the rest of the file constitutes the example
            answers: List[str] = []
            for example in examples:
                answers.append(example.split("So the answer is ")[-1].removesuffix("."))  # extract answer from example
            full_examples: List[str] = []
```